forwarder/kafka/kafka_helpers.py

```diff
@@ -29,25 +63,47 @@ def create_producer(
     )
 
 
-def create_consumer(broker_address: str) -> Consumer:
-    return Consumer(
-        {
-            "bootstrap.servers": broker_address,
-            "group.id": uuid.uuid4(),
-            "default.topic.config": {"auto.offset.reset": "latest"},
-        }
+def create_consumer(
+    broker_address: str,
+    sasl_mechanism: Optional[str] = None,
+    username: Optional[str] = None,
+    password: Optional[str] = None,
+) -> Consumer:
+    consumer_config = {
+        "bootstrap.servers": broker_address,
+        "group.id": uuid.uuid4(),
+        "default.topic.config": {"auto.offset.reset": "latest"},
+    }
+    if sasl_mechanism:
+        consumer_config.update(get_sasl_config(sasl_mechanism, username, password))
+    return Consumer(consumer_config)
+
+
+def get_broker_topic_and_username_from_uri(uri: str) -> Tuple[str, str, str, str]:
+    topic = uri.split("/")[-1]
+    if "/" not in uri or not topic:
+        raise RuntimeError(
+            f"Unable to parse URI {uri}, should be of form [[SASL_MECHANISM\\]username@]localhost:9092/topic"
+        )
+    broker_and_username = "".join(uri.split("/")[:-1])
+    broker, sasl_mechanism, username = get_broker_and_username_from_uri(
+        broker_and_username
     )
+    return broker, topic, sasl_mechanism, username
 
 
-def get_broker_and_topic_from_uri(uri: str) -> Tuple[str, str]:
```

Comment:
rename to a generic name

Reply:
One attempt tried [here](https://github.com/ess-dmsc/forwarder/pull/173/commits/70ec2f50d414dfbbd9e3cc7aebed25b89d6e6664), but the code it's uglier than before.
I lean towards refactoring the URI parsing so it is the calling code (and not the parser) who decides which fields are mandatory.

That is however outside the scope of this PR: https://jira.esss.lu.se/browse/ECDC-3066